src/Microsoft.AspNetCore.Server.Kestrel/Internal/Networking/UvWriteReq.cs

```diff
@@ -200,31 +200,36 @@ public LibuvAwaitable<UvWriteReq> WriteAsync(UvStreamHandle handle, ArraySegment
             {
                 _callback = null;
                 _state = null;
-                Unpin(this);
+                UnpinGcHandles();
                 throw;
             }
         }
 
-        private static void Unpin(UvWriteReq req)
+        // Safe handle has instance method called Unpin
+        // so using UnpinGcHandles to avoid conflict 
+        private void UnpinGcHandles()
         {
-            foreach (var pin in req._pins)
+            var pinList = _pins;
```

Comment:
so its operating on a local register pointer of the List rather than operating on memory location